projects/gateway2/translator/listener/gateway_listener_translator.go

```diff
@@ -786,3 +788,46 @@ func makeVhostName(
 ) string {
 	return utils.SanitizeForEnvoy(ctx, parentName+"~"+domain, "vHost")
 }
+
+// AppendHealthCheckListener appends a readiness listener with the health check plugin defined on it
+// and returns the updated listener list
+func AppendHealthCheckListener(listeners []*v1.Listener) []*v1.Listener {
+	return append(listeners, GenerateHealthCheckListener())
+}
+
+// GenerateHealthCheckListener returns a readiness listener with the health check plugin defined on it
+func GenerateHealthCheckListener() *v1.Listener {
```

Comment:
Shouldnt we validate that this is only added if configured? Whats the impact if we expose this and someone previously didnt expose it?

Reply:
Adding validation might slow translation since it would need to fetch the GWP and check it during every translation cycle. Also I don't think we trigger translation when the gwp changes
The only impact I can see would be that the health check endpoint would be exposed although not via a service